prov/verbs/src/verbs_info.c

```diff
@@ -1170,22 +1175,32 @@ int fi_ibv_init_info(const struct fi_info **all_infos)
 	struct ibv_context **ctx_list;
 	struct fi_info *fi = NULL, *tail = NULL;
 	const struct verbs_ep_domain *ep_type[VERBS_NUM_DOMAIN_TYPES];
-	int ret = 0, i, j, num_devices;
+	int ret = 0, i, j, num_devices, dom_count = 0;
+	enum fi_log_level level = FI_LOG_INFO;
 
 	*all_infos = NULL;
 
 	/* List XRC MSG_EP domain before default RC MSG_EP if requested */
-	if (fi_ibv_gl_data.msg.prefer_xrc) {
-		ep_type[0] = &verbs_msg_xrc_domain;
-		ep_type[1] = &verbs_msg_domain;
+	if (!VERBS_HAVE_XRC) {
+		if (fi_ibv_gl_data.msg.prefer_xrc)
+			level = FI_LOG_WARN;
+		FI_LOG(&fi_ibv_prov, level, FI_LOG_FABRIC,
+		       "XRC not built into provider, skip allocating fi_info "
+		       "for XRC FI_EP_MSG endpoints\n");
```

Comment:
I think it's safe to avoid logging a message that says we're not doing something

Reply:
This change maintains part of the previous logging behavior (at "info" level). I can modify it to just log it as warning when "prefer xrc" is requested. The user would want to know why XRC is unavailable when it was requested.